src/m_Do/m_Do_Reset.cpp

```diff
@@ -1,32 +1,118 @@
-//
-// Generated by dtk
-// Translation Unit: m_Do_Reset.cpp
-//
+/**
+ * m_Do_Reset.cpp
+ * Game Reset Management
+ */
 
 #include "m_Do/m_Do_Reset.h"
-#include "dolphin/types.h"
+#include "JSystem/JUtility/JUTGamePad.h"
+#include "JSystem/JUtility/JUTXfb.h"
+#include "dolphin/gx/GX.h"
+#include "dolphin/os/OS.h"
+#include "m_Do/m_Do_DVDError.h"
+#include "m_Do/m_Do_audio.h"
 
 /* 80017D40-80017D44       .text my_OSCancelAlarmAll__Fv */
-void my_OSCancelAlarmAll() {
-    /* Nonmatching */
-}
+static void my_OSCancelAlarmAll() {}
 
 /* 80017D44-80017D7C       .text destroyVideo__Fv */
 void destroyVideo() {
-    /* Nonmatching */
+    JUTVideo::destroyManager();
+    GXSetDrawDoneCallback(NULL);
+    VISetBlack(1);
+    VIFlush();
+    VIWaitForRetrace();
 }
 
+#if VERSION == VERSION_JP
+int mDoRst::mResetFlag;
+int mDoRst::mResetPrepare;
+int mDoRst::m3ButtonResetFlag;
+int mDoRst::m3ButtonResetPort = -1;
+#else
+mDoRstData* mDoRst::mResetData;
+#endif
+
 /* 80017D7C-80017E40       .text mDoRst_reset__FiUli */
-void mDoRst_reset(int, unsigned long, int) {
-    /* Nonmatching */
+void mDoRst_reset(int param_0, unsigned long param_1, int param_2) {
+    if (DVDGetDriveStatus() == DVD_STATE_FATAL_ERROR) {
+        do {
+        } while (true);
+    }
+
+    JUTXfb::getManager()->clearIndex();
+    mDoDvdErr_ThdCleanup();
+    JUTGamePad::clearForReset();
+    mDoAud_zelAudio_c::offInitFlag();
+
+#if VERSION != VERSION_JP
```

Comment:
I wonder if we should start doing `VERSION > VERSION_JP` to make it clear things were added in future versions. Or `VERSION >= VERSION_USA` might be most clear?

Reply:
the 2nd one might be more clear, but yeah it should probably be discussed more in the discord along with how best to handle these kinds of version differences, since ifdef hell could get bad